### ISSUE TITLE
Allow support for fibers v4 and v5

### ISSUE THREAD

**Issue by christian-bromann** [webdriverio/webdriverio#5364]
**Is your feature request related to a problem? Please describe.**
Fibers got updated to v5 to support Node.js v14. Updating it would break support for Node.js v10. Also the fibers maintainer has announced to not further maintain the package which leads to the question how we want to further provide support for synchronous execution.

Any help or feedback is appreciated.

**Comment by Abel0916** [comment]
Encountered this problem when I upgrade Fibers to V5. Then WDIO sync way doesn't work.

**Comment by pmerwin** [comment]
Yes, I think we ran into an issue in AWS Linux 2 lambda, fibers vs 5, and sync mode exits with a code 1 immediately on test execution but only in AWS lambda, could not reproduce locally on OSX.

**Comment by christian-bromann** [comment]
@Abel0916 don't update to Fibers v5. It seems that Fibers in v4 still works with the latest Node v14. Therefor there is no need of action right now.

**Comment by pathmapper** [comment]
> It seems that Fibers in v4 still works with the latest Node v14

@christian-bromann on `linux` and `macos` it seems to work, but on `windows` it's failing: 

https://github.com/maputnik/editor/pull/695

https://github.com/maputnik/editor/pull/695/checks?check_run_id=715807535

**Comment by christian-bromann** [comment]
@pathmapper thanks for the feedback, I will prioritise this then.

**Comment by pathmapper** [comment]
@christian-bromann our failing test was related to `node-sass` (https://github.com/maputnik/editor/pull/695#issuecomment-635209469). Sorry for the noise.

**Comment by christian-bromann** [comment]
ahh ok, nevermind then 😆

**Comment by hexsprite** [comment]
is it recommended to use async then for new projects? or you intend to support sync regardless of fibers support?

**Comment by christian-bromann** [comment]
We are trying to continue supporting sync mode but we can't promise it.

**Comment by thomaseitler** [comment]
Hi,

sorry if this is not related at all - I can create a new issue if necessary. However the error mentions the fibers module.

I just updated to node ```v15.2.1``` and cannot complete the initial config ``` npx wdio config -y ```

Infor from the debug log:

```
...
2016 verbose stack Error: fibers@4.0.3 install: `node build.js || nodejs build.js`
2016 verbose stack spawn ENOENT
2016 verbose stack     at ChildProcess.<anonymous> (/usr/lib/node_modules/npm/node_modules/npm-lifecycle/lib/spawn.js:48:18)
2016 verbose stack     at ChildProcess.emit (node:events:329:20)
2016 verbose stack     at maybeClose (node:internal/child_process:1055:16)
2016 verbose stack     at Process.ChildProcess._handle.onexit (node:internal/child_process:288:5)
2017 verbose pkgid fibers@4.0.3
2018 verbose cwd /home/user/Automated Testing/wdio-test
2019 verbose Linux 5.8.18-1-MANJARO
2020 verbose argv "/usr/bin/node" "/bin/npm" "install" "--save-dev" "@wdio/local-runner" "@wdio/mocha-framework" "@wdio/spec-reporter" "wdio-chromedriver-service" "@wdio/sync" "chromedriver"
2021 verbose node v15.2.1
2022 verbose npm  v6.14.8
2023 error code ELIFECYCLE
2024 error syscall spawn
2025 error file sh
2026 error errno ENOENT
2027 error fibers@4.0.3 install: `node build.js || nodejs build.js`
2027 error spawn ENOENT
2028 error Failed at the fibers@4.0.3 install script.
2028 error This is probably not a problem with npm. There is likely additional logging output above.
2029 verbose exit [ 1, true ]
```

Error from the console:
```
Error: something went wrong during setup: null
    at runConfig (/home/user/Automated Testing/wdio-test/node_modules/@wdio/cli/build/commands/config.js:61:15)
    at async Object.handler (/home/user/Automated Testing/wdio-test/node_modules/@wdio/cli/build/commands/config.js:111:9)
    at Object.handler (/home/user/Automated Testing/wdio-test/node_modules/@wdio/cli/build/commands/config.js:114:15)
```
I also tried installing fibers directly and got the same error for the newest version.
```
npm ERR! code ELIFECYCLE
npm ERR! syscall spawn
npm ERR! file sh
npm ERR! errno ENOENT
npm ERR! fibers@5.0.0 install: `node build.js || nodejs build.js`
npm ERR! spawn ENOENT
npm ERR! 
npm ERR! Failed at the fibers@5.0.0 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
```

So this is an issue with fibers (https://github.com/laverdet/node-fibers) itself and should be fixed there? Is there any alternative besides downgrading to node v14 again?

Thanks!

**Comment by L0stSoul** [comment]
Hey guys! at least for now can fiber dependency be updated to version 5? version 5 contains pre-built binaries for latest node LTS which allow to not install g++, make and other stuff :) which make life way easier.

**Comment by christian-bromann** [comment]
@L0stSoul according to the readme:

> If you are running nodejs v10.x then you will need to use npm install fibers@4. Older versions of nodejs will require older and older version of fibers.

As we still support Node v10 we can't make an update just yet. However we do plan a release soon and will drop support for Node v10 which allows us to tackle this ticket.

**Comment by christian-bromann** [comment]
Fixed with #6236
